### PR TITLE
[SYCLomatic] Change default engine from philox4x32x10 to mcg59 for both host and device generators

### PR DESCRIPTION
**Pull request by zhiweij1**
Signed-off-by: Jiang, Zhiwei <zhiwei.jiang@intel.com>

### CODE REVIEW

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Review by ziranzha — APPROVED** [review]

File: clang/test/dpct/curand-device2.cu [hunk 1]
```diff
@@ -195,7 +195,10 @@ __global__ void kernel3() {
 
   //CHECK:oneapi::mkl::rng::device::skip_ahead(rng1.get_engine(), {0, 1 * (std::uint64_t(1) << 63)});
   //CHECK-NEXT:oneapi::mkl::rng::device::skip_ahead(rng2.get_engine(), {0, static_cast<std::uint64_t>(2 * 4)});
-  //CHECK-NEXT:oneapi::mkl::rng::device::skip_ahead(rng3.get_engine(), {0, static_cast<std::uint64_t>(3 * 8)});
+  //CHECK-NEXT:/*
+  //CHECK-NEXT:DPCT1105:{{[0-9]+}}: The mcg59 random number generator is used. The subsequence argument is treated as 0. You need to verify the migration.
+  //CHECK-NEXT:*/
+  //CHECK-NEXT:oneapi::mkl::rng::device::skip_ahead(rng3.get_engine(), 0);
```
Comment (zhimingwang36) [inline clang/test/dpct/curand-device2.cu]: call to oneapi::mkl::rng::device::skip_ahead() is redudant

File: clang/test/dpct/curand-device.cu [hunk 2]
```diff
@@ -152,8 +152,11 @@ int foo() {
   return 0;
 }
 
-//CHECK:void kernel(dpct::rng::device::rng_generator<oneapi::mkl::rng::device::philox4x32x10<1>> *state) {
-//CHECK-NEXT:  *state = dpct::rng::device::rng_generator<oneapi::mkl::rng::device::philox4x32x10<1>>(1111, {0, 2222 * 8});
+//CHECK:void kernel(dpct::rng::device::rng_generator<oneapi::mkl::rng::device::mcg59<1>> *state) {
+//CHECK-NEXT:  /*
+//CHECK-NEXT:  DPCT1105:{{[0-9]+}}: The mcg59 random number generator is used. The subsequence argument is treated as 0. You need to verify the migration.
```
Comment (zhimingwang36) [inline clang/test/dpct/curand-device.cu]:
```suggestion
//CHECK-NEXT:  DPCT1105:{{[0-9]+}}: The mcg59 random number generator is used. The subsequence argument "2222" is ignored.  You need to verify the migration.
```

**Comment by iMartyan** [comment]
Thank you for making the transaction to mcg59 engine!
Just another moment for this change: default distribution for mcg59 directly mapping to curand() shall be dispatched to ...rng::device::uniform_bits<std::uint32_t>. "Bits" distribution is supported but in case of mcg59 it would return some guaranteed zero bits, so "uniform_bits" is needed.

**Comment by zhiweij1** [comment]
The failed case features/ccl is not related to this PR.

**Comment by zhiweij1** [comment]
Hi @iMartyan , I have updated this PR according to your feedback.

**Comment by iMartyan** [comment]
Hi @zhiweij1,
Everything looks good to me.
Thank you for your work!

File: clang/lib/DPCT/APINamesCURAND.inc [hunk 3]
```diff
@@ -40,15 +40,7 @@ CONDITIONAL_FACTORY_ENTRY(
...
-        FEATURE_REQUEST_FACTORY(
-            HelperFeatureEnum::RngUtils_rng_generator_get_engine,
-            WARNING_FACTORY_ENTRY("skipahead_sequence",
-            CALL_FACTORY_ENTRY(
-                "skipahead_sequence",
-                CALL("oneapi::mkl::rng::device::skip_ahead",
-                     MEMBER_CALL(DEREF(makeDerefArgCreatorWithCall(1)),
-                                        false, "get_engine"),
-                     ARG("0"))), Diagnostics::CHECK_SEMANTICS_OF_DIFFERENT_GENERATOR)))),
+        REMOVE_API_FACTORY_ENTRY_WITH_MSG("skipahead_sequence", "this API is redudant for mcg59 engine."))),
```
Comment (zhimingwang36) [inline clang/lib/DPCT/APINamesCURAND.inc]: keep align with L68.

Comment (zhiweij1) [reply clang/lib/DPCT/APINamesCURAND.inc]: fixed

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**